kratos/utilities/result_dabatase.cpp

```diff
@@ -0,0 +1,216 @@
+const EntityDatabase& VariableDatabase::GetEntityData(const IndexType EntityIndex) const
+{
+    KRATOS_DEBUG_ERROR_IF(EntityIndex == mData.size()) << "Incompatible size. EntityIndex: " << EntityIndex << ". Size: " << mData.size() << std::endl;
+    return mData[EntityIndex];
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+double VariableDatabase::GetValue(
+    const IndexType EntityIndex,
+    const double Time,
+    const SizeType ComponentIndex,
+    const SizeType GPIndex
+    ) const
+{
+    KRATOS_DEBUG_ERROR_IF(EntityIndex == mData.size()) << "Incompatible size. EntityIndex: " << EntityIndex << ". Size: " << mData.size() << std::endl;
+    return mData[EntityIndex].GetValue(Time, ComponentIndex, GPIndex);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void VariableDatabase::SetValues(
+    const Vector& rValuesX,
+    const Vector& rValuesY,
+    const IndexType EntityIndex,
+    const SizeType ComponentIndex,
+    const SizeType GPIndex
+    )
+{
+    auto& r_database = mData[EntityIndex];
+    r_database.SetValues(rValuesX, rValuesY, ComponentIndex, GPIndex);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void VariableDatabase::Clear()
+{
+    // Clear stored databases
+    for (auto& r_database : mData) {
+        r_database.Clear();
+    }
+
+    // Clear the container
+    mData.clear();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ResultDatabase::Initialize(
+    const std::vector<IndexType>& rVariablesIndexes,
+    const std::vector<IndexType>& rValuesSizes,
+    const SizeType NumberOfEntites,
+    const SizeType NumberOfGP
+    )
+{
+    // If no variables we skip
+    if (rVariablesIndexes.size() == 0) {
+        return void();
+    }
+
+    KRATOS_ERROR_IF_NOT(rVariablesIndexes.size() == rValuesSizes.size()) << "Inconsistent sizes in the values sizes and the variable indexes" << std::endl;
+
+    // Auxiliar lambda to generate vectors of tables
+    auto table_generator =[](const SizeType NumberOfEntites, const SizeType NumberOfComponents, const SizeType NumberOfGP){
+        EntityDatabase::GPDatabaseType aux_1(NumberOfComponents, nullptr);
+        EntityDatabase aux_2(NumberOfGP, aux_1);
+        VariableDatabase data(NumberOfEntites, aux_2);
+        for (IndexType k = 0; k < NumberOfEntites; ++k){
+            for (IndexType j = 0; j < NumberOfGP; ++j){
+                for (IndexType i = 0; i < NumberOfComponents; ++i){
+                    data[k][j][i] = new Table<double, double>();
```

Comment:
I thought you used a `shared_ptr` now?

Reply:
I reverted as I said

Reply:
(raw are faster)

Reply:
I found the way to delete safely

Reply:
I still don't understand why this makes such a performance impact & why a smart pointer cannot be used, but I also don't want to discuss it any more ...

Reply:
I used a smart pointer originally, but a raw pointer does brbrbrbrbr

![imagen](https://user-images.githubusercontent.com/19991680/83762737-24171400-a678-11ea-99de-ed96b54a45fa.png)